overlord/snapstate/snapstate_test.go

```diff
@@ -12080,8 +12093,8 @@ func (s *snapmgrTestSuite) TestUpdateLayoutsChecksFeatureFlag(c *C) {
 	_, err := snapstate.Update(s.state, "some-snap", "channel-for-layout", snap.R(0), s.user.ID, snapstate.Flags{})
 	c.Assert(err, ErrorMatches, "experimental feature disabled - test it by setting 'experimental.layouts' to true")
 
-	// enable layouts
-	tr := config.NewTransaction(s.state)
+	// When layouts are enabled we can refresh to a snap depending on the feature.
```

Comment:
I think we're missing a test for the update path but with layouts feature using the default. That would be the case of a snap that did not use layouts before but got them now. It's probably enough to just test the `Update()` as the checks are in `doUpdate()` which is called by `UpdateMany()` too.